Week3/queries.js

```diff
@@ -0,0 +1,62 @@
+//SELECT QUERIES
+const listsAndToDosByUserId = `SELECT lists.name, todos.description
+FROM lists
+INNER JOIN lists_todos ON lists.user_id = ? AND lists_todos.list_id = lists.id
+LEFT JOIN todos ON lists_todos.todo_id= todos.id`;
+
+const listsByUserId = `SELECT name
+FROM lists JOIN users ON lists.user_id = users.id AND users.id = ?`;
+
+const remindersByUserId = `SELECT name, reminder from lists 
+WHERE user_id = ? AND reminder IS NOT NULL
+`;
+
+const allDoneToDosByUserId = `SELECT description FROM todos WHERE id IN (
+  SELECT todo_id FROM lists_todos WHERE is_done = ? AND list_id IN(
+    SELECT id FROM lists WHERE user_id = ?
+  )
+)`;
+
+const s_Id_Todos_Desc = 'select id from todos where description = ?';
```

Comment:
this naming practice is really outdated. It is extremely hard to read as I got to experience first hand. Also I would have grouped things per entity rather than request type. Here is an alternative way to organize this file that is much easier to read and understand:
```
const queries = {
 user: {
     findByName: 'SELECT id FROM users WHERE username = ?',
     delete: 'DELETE ...'
  }
  todo: {
      findById: 'SELECT ...'
      insert: 'INSERT into todos set'
      ....
   },
   ...
}
module.export(queries);
```
Then when you are using them:
```
const queries = require('/queries.js');
execQuery(queries.todos.findById, [todoId]);
```
Do you agree that this is this easier to read?

Reply:
yes of course. this one is clear, understandable and readable for sure. thanks.